groups/bsl/bslstl/bslstl_ratio.t.cpp

```diff
@@ -0,0 +1,383 @@
+// ----------------------------------------------------------------------------
+
+// ============================================================================
+//                         OTHER MACROS
+// ----------------------------------------------------------------------------
+
+//=============================================================================
+//                  GLOBAL TYPEDEFS/CONSTANTS FOR TESTING
+//-----------------------------------------------------------------------------
+static bool verbose;
+static bool veryVerbose;
+static bool veryVeryVerbose;
+static bool veryVeryVeryVerbose;
+
+//=============================================================================
+//                  GLOBAL HELPER FUNCTIONS FOR TESTING
+//-----------------------------------------------------------------------------
+
+//=============================================================================
+//                              MAIN PROGRAM
+//-----------------------------------------------------------------------------
+
+int main(int argc, char *argv[])
+{
+    int            test = argc > 1 ? atoi(argv[1]) : 0;
+                verbose = argc > 2;
+            veryVerbose = argc > 3;
+        veryVeryVerbose = argc > 4;
+    veryVeryVeryVerbose = argc > 5;
+
+    printf("TEST " __FILE__ " CASE %d\n", test);
+
+    switch (test) { case 0:  // Zero is always the leading case.
+      case 1: {
+        // --------------------------------------------------------------------
+        // TESTING <ratio> HEADER VERIFICATION
+        //
+        // Concerns:
+        //: 1 That the operations are successfully imported into the 'bsl'
+        //:   namespace from the underlying STL implementation.
+        //: 
+        //: 2 That the short versions of the comparison operators, such
+        //:   as 'ratio_less_v' exist and work even if the underlying STL
+        //:   does not provide them.
+        //
+        // Plan:
+        //: 1 Verify that the template 'ratio' exists in the bsl namespace and
+        //:   that it behaves like std::ratio.
+        //:
+        //: 2 Verify that the SI typedefs exist and have the correct numerator
+        //:   and denominator.
+        //:
+        //: 3 Verify that the comparison operations exist, and give expected
+        //:   results. Check again for the '_v' versions.
+        //:
+        //: 4 Verify that the arithmetic operations exist, and give expected
+        //:   results.
+        //
+        // Testing:
+        //  Basic ratio functionality - numerator/denominator/nested 'type'
+        //   SI prefixes - kilo, mega, milli, etc.
+        //   Ratio comparisons - less/greater/equal, etc.
+        //   Ratio operations - add/subtract/multiply/divide, etc.
+        // --------------------------------------------------------------------
+
+        if (verbose) printf("\nTesting <ratio> HEADER VERIFICATION"
+                            "\n===================================\n");
+
```

Comment:
See bslstl_chrono.t.cpp CASE 1 for how to allow this test driver to also build on pre-C++11 platforms.  Note that under the `#else` I would use `verbose` instead of `veryVerbose`.  (BTW, bslstl_chrono.t.cpp reasonably illustrates correspondence of "Testing:" and "TEST PLAN" (except in that test driver it should be "[ 1]", "[ 2]" rather than "[1]", "[2]").)